addons/binding/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/handler/YeelightHandlerBase.java

```diff
@@ -265,4 +279,16 @@ void updateBrightnessAndColorUI(DeviceStatus status) {
                     new PercentType((status.getCt() - COLOR_TEMPERATURE_MINIMUM) / COLOR_TEMPERATURE_STEP));
         }
     }
+
+    int getDuration() {
+        // Duration should not be null, but just in case do a null check.
+        try {
+            return getThing().getConfiguration().get(YeelightBindingConstants.PARAMETER_DURATION) == null ? 500
+                    : ((Number) getThing().getConfiguration().get(YeelightBindingConstants.PARAMETER_DURATION))
+                            .intValue();
+        } catch (Exception e) {
```

Comment:
The exception I was trying to catch is casting error. I am unsure if there will ever be a case when the cast getThing() to a Number will fail since, in the config XML, it's set as an integer.

Maybe we can remove the whole try/catch?

Reply:
I think we can omit the try-catch-block in this case. The validation of the configuration will be taken care of by the framework and we have a default value.

```java
Integer duration = (Integer) getThing().getConfiguration().get(PARAMETER_DURATION);
return duration == null ? 500 : duration.intValue();
```

Reply:
Done. Thanks.